mxcube3/routes/signals.py

```diff
@@ -137,14 +137,13 @@ def loaded_sample_changed(sample):
         barcode = ''
 
     logging.getLogger("HWR").info('loaded sample changed now is: ' + address)
-
-    if isinstance(sample, Pin):
+ 
+    try:
```

Comment:
why did you remove this? I think it could also be ~Container

Reply:
The code as it were before generated errors when executed, since sample_data was undefined if
sample was not a Pin

And I dont understand why it would be container, the event is sample changed, so sample should 
not be a container. For me it indicates the the event is called when its not supposed to be called

If its the case it should just return at the top of the function, because all the logic in this function corresponds to a sample beeing loaded and is not valid for a container.

Reply:
Maybe I am mixing containers/baskets/pins etc...

This code was working with mockups and the Pin was needed (for some error I cannot remember), but I see and agree with your point, so we can put everything on the same `if`

Reply:
this is a merge problem, the `try` should be inside the `if`